daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2264,6 +2264,9 @@ func (devices *DeviceSet) unmountAndDeactivateAll(dir string) {
 		// and the device will be released when that container dies.
 		if err := unix.Unmount(fullname, unix.MNT_DETACH); err != nil && err != unix.EINVAL {
 			logger.Warnf("Shutdown unmounting %s, error: %s", fullname, err)
+		} else if err == nil {
+			logger.Debugf("Remove %s", fullname)
+			os.RemoveAll(fullname)
```

Comment:
I had a very quick look at other code paths after this; would this affect (e.g.) deferred removal later on?

Also, this should probably be `os.Remove()`, not `os.RemoveAll()`

Reply:
> I had a very quick look at other code paths after this; would this affect (e.g.) deferred removal later on?
The directory shall be empty after umount, remove the empty dir not affect deferred removal.

> Also, this should probably be os.Remove(), not os.RemoveAll()
os.Remove() not work when load/pull images, the files can be extracted to the umounted directory just when umount done, os.Remove() can not remove the directory when it has dirty files.